lib/src/components/modal_main_page.dart

```diff
@@ -44,22 +45,14 @@ class _ModalMainPageState extends State<ModalMainPage> {
                 children: [
                   CupertinoSlidingSegmentedControl<int>(
                     groupValue: _groupValue,
-                    onValueChanged: (value) => _groupValue = value,
+                    onValueChanged: (value) => setState(() {
+                      _groupValue = value;
+                    }),
                     backgroundColor: Colors.grey.shade300,
                     padding: const EdgeInsets.all(4),
                     children: {
-                      0: _Segment(
-                        text: 'Mobile',
-                        onTap: () => setState(
-                          () => _groupValue = 0,
-                        ),
-                      ),
-                      1: _Segment(
-                        text: 'QR Code',
-                        onTap: () => setState(
-                          () => _groupValue = 1,
-                        ),
-                      ),
+                      0: Utils.isDesktop() ? qrSegment() : listSegment(),
```

Comment:
This part is a little bit hacky, but it allows the QR to be displayed first on desktop (imitating the JS implementation). Display order is determined by map order and not the key.